/*
  CarStandFCOOL API

  An API for managing a cloud-based car dealership service

  The version of the OpenAPI document: 1.0.0

  Generated by OpenAPI Generator: https://openapi-generator.tech
*/

syntax = "proto3";

package openapitools.services.inspectionservice;

import "google/protobuf/empty.proto";
import public "models/inspection.proto";

service InspectionService {
  rpc InspectionCreate (InspectionCreateRequest) returns (Inspection);

  rpc InspectionDelete (InspectionDeleteRequest) returns (google.protobuf.Empty);

  rpc InspectionReadAll (google.protobuf.Empty) returns (InspectionReadAllResponse);

  rpc InspectionReadOne (InspectionReadOneRequest) returns (Inspection);

  rpc InspectionUpdate (InspectionUpdateRequest) returns (Inspection);

}

message InspectionCreateRequest {
  // Inspection to create
  Inspection inspection = 1;

}

message InspectionDeleteRequest {
  // ID of the inspection
  int32 inspectionId = 1;

}

message InspectionReadAllResponse {
  repeated Inspection data = 1;
}

message InspectionReadOneRequest {
  // ID of the inspection
  int32 inspectionId = 1;

}

message InspectionUpdateRequest {
  // ID of the inspection
  int32 inspectionId = 1;
  Inspection inspection = 2;

}

